src/io/ioexternalbus.cpp

```diff
@@ -4,13 +4,30 @@
 #include <QPainter>
 #include <QPen>
 
+#include <QtWidgets>
+
 #include "STLExtras.h"
 #include "ioregistry.h"
 
+#ifndef _MSC_VER
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#endif
+
+#define dprintf \
```

Comment:
As above, we should use some forwarding to UI instead.

Reply:
This dprintf macro is for debugging only and will be removed in the final version.